esp/services/ws_fs/ws_fsService.cpp

```diff
@@ -1842,13 +1840,31 @@ void CFileSprayEx::readAndCheckSpraySourceReq(MemoryBuffer& srcxml, const char*
                 sourcePath.append(srcPath).trim();
             }
             const char * hostGroup = dropZone->queryProp("@hostGroup");
-            const char * host = "localhost"; // by default assume storage plane mounted on dfuserver
             if (hostGroup)
             {
                 Owned<IPropertyTree> match = getHostGroup(hostGroup,true);
-                host = match->queryProp("hosts[1]");
+                if (!isEmptyString(srcIP))
+                {
+                    Owned<IPropertyTreeIterator> hostIter = match->getElements("hosts");
+                    ForEach (*hostIter)
+                    {
+                        const char *knownIP = hostIter->query().queryProp(nullptr);
+                        if (strcmp(knownIP, srcIP)==0)
+                        {
+                            sourceIPReq.set(knownIP);
+                            return;
```

Comment:
should this be a break; , to fall through so that sourcePathReq is filled in?

Reply:
agreed